file/src/main/scala/akka/stream/alpakka/file/impl/archive/ZipReaderSource.scala

```diff
@@ -0,0 +1,100 @@
+/*
+ * Copyright (C) 2016-2020 Lightbend Inc. <https://www.lightbend.com>
+ */
+
+package akka.stream.alpakka.file.impl.archive
+
+import akka.NotUsed
+import akka.annotation.InternalApi
+import akka.stream.{Attributes, Outlet, SourceShape}
+import akka.stream.scaladsl.Source
+import akka.stream.stage.{GraphStage, GraphStageLogic, OutHandler}
+import akka.util.ByteString
+
+import java.io.{File, FileInputStream}
+import java.util.zip.{ZipEntry, ZipInputStream}
+
+case class ZipArchiveMetadata(name: String)
```

Comment:
This class is user-facing API and should preferably live in `akka.stream.alpakka.file` instead. Add a `getName` for Java folks.

Reply:
Done!